### PR TITLE
fix(l1): receipts invalid size libmdbx

### PR DESCRIPTION
**Pull request by MarcosNicolau**
**Motivation**
Syncing holesky.

**Description**
While syncing on Holesky, we encountered blocks that exceeded the maximum allowed entry size in libmdbx. Increasing the page size was not a viable solution, as even the maximum allowable size in libmdbx wasn't enough for some blocks.

To address this, we implemented an `IndexedChunk` trait that splits oversized data into smaller chunks, ensuring each piece stays within the maximum value size limit. Since libmdbx is structured as a B+ tree map, we had to ensure correct order of chunks. We achieved this by appending a byte representing the chunk index, this way we preserve the correct sequence of chunks during retrieval.

Moving forward, we should consider whether other tables might also exceed this size limit. If so, we can extend this trait.

Closes None

### CODE REVIEW

**Review by mpaulucci — APPROVED** [review]
great work!

File: crates/storage/store/store_db/libmdbx.rs [hunk 1]
```diff
@@ -509,15 +517,13 @@ impl StoreEngine for Store {
...
-            receipts.push(encoded_receipt);
+        while let Some(receipt) = IndexedChunk::read_from_db(&mut cursor, key)? {
+            receipts.push(receipt);
             receipt_index += 1;
             key = (*block_hash, receipt_index).into();
         }
 
-        Ok(receipts.into_iter().map(|receipt| receipt.to()).collect())
+        Ok(receipts)
     }
```
Comment (Oppen) [inline crates/storage/store/store_db/libmdbx.rs:527]: I think there might be a risk of infinite looping here. If your block uses 256 chunks then after chunk with index `255` the addition overflows, and you'll ask for chunk with index `0` of the same hash, and thus start the whole cycle again forever.

Comment (Oppen) [reply crates/storage/store/store_db/libmdbx.rs:527]: Maybe we should store the size somewhere.

Comment (MarcosNicolau) [reply crates/storage/store/store_db/libmdbx.rs:527]: Yes, I considered that as well. The size is defined as a u8, meaning we can have up to 256 chunks. However, if each value is 2022 bytes, we'd need to store around 517,632 bytes (~0.5 MB) to reach this limit. Tbh, I am not sure about the log size in ethereum and whether a receipt can take that amount of space. 

For now, as we discussed offline, we'll keep it as is and address any limitations if they arise. I'll also update the code to explicitly reflect this limit and leave a clear warning.

Comment (MarcosNicolau) [reply crates/storage/store/store_db/libmdbx.rs:532]: Solved in [18b2941](https://github.com/lambdaclass/ethrex/pull/2065/commits/18b294158986c67f750dadbbb9e044c0d2d1a4bc).

**Review by avilagaston9 — APPROVED** [review]
Works for me in the server! Let's wait for @Oppen's approval anyways

**Review by Oppen — APPROVED** [review]
Looks good. Just to be on the safe side, can we add a test for a write-then-read that takes exactly 256 chunks?

**Comment by MarcosNicolau** [comment]
> Looks good. Just to be on the safe side, can we add a test for a write-then-read that takes exactly 256 chunks?

Done in [bd61db0](https://github.com/lambdaclass/ethrex/pull/2065/commits/bd61db0d0ac90db6bf555b9da34d447536f1396c). I’ve updated the existing test to include this case. Now, we verify the boundary condition at exactly 256 chunks, as well as an edge case that exceeds it by a single byte.
